staging/src/k8s.io/kubectl/pkg/util/completion/completion_test.go

```diff
@@ -330,57 +330,76 @@ func TestPodResourceNameCompletionFuncJointFormTooManyArgs(t *testing.T) {
 	checkCompletion(t, comps, []string{}, directive, cobra.ShellCompDirectiveNoFileComp)
 }
 
-func TestPodResourceNameAndContainerCompletionFuncNoArgsPodName(t *testing.T) {
-	tf, cmd := prepareCompletionTest()
-	pods, _, _ := cmdtesting.TestData()
-	addResourceToFactory(tf, pods)
-
-	compFunc := PodResourceNameAndContainerCompletionFunc(tf)
-	comps, directive := compFunc(cmd, []string{}, "b")
-	checkCompletion(t, comps, []string{"bar"}, directive, cobra.ShellCompDirectiveNoFileComp)
-}
-
-func TestPodResourceNameAndContainerCompletionFuncNoArgsResources(t *testing.T) {
-	tf, cmd := prepareCompletionTest()
-	pods, _, _ := cmdtesting.TestData()
-	addResourceToFactory(tf, pods)
-
-	compFunc := PodResourceNameAndContainerCompletionFunc(tf)
-	comps, directive := compFunc(cmd, []string{}, "s")
-	checkCompletion(
-		t, comps, []string{"services/", "statefulsets/"},
-		directive, cobra.ShellCompDirectiveNoFileComp|cobra.ShellCompDirectiveNoSpace)
-
-}
-
-func TestPodResourceNameAndContainerCompletionFuncTooManyArgs(t *testing.T) {
-	tf, cmd := prepareCompletionTest()
-	pods, _, _ := cmdtesting.TestData()
-	addResourceToFactory(tf, pods)
-
-	compFunc := PodResourceNameAndContainerCompletionFunc(tf)
-	comps, directive := compFunc(cmd, []string{"pod-name", "container-name"}, "")
-	checkCompletion(t, comps, []string{}, directive, cobra.ShellCompDirectiveNoFileComp)
-}
-
-func TestPodResourceNameAndContainerCompletionFuncJointFormNoArgs(t *testing.T) {
-	tf, cmd := prepareCompletionTest()
-	pods, _, _ := cmdtesting.TestData()
-	addResourceToFactory(tf, pods)
-
-	compFunc := PodResourceNameAndContainerCompletionFunc(tf)
-	comps, directive := compFunc(cmd, []string{}, "pod/b")
-	checkCompletion(t, comps, []string{"pod/bar"}, directive, cobra.ShellCompDirectiveNoFileComp)
-}
+func TestResourceAndContainerNameCompletionFunc(t *testing.T) {
+	barPod := getTestPod()
 
-func TestPodResourceNameAndContainerCompletionFuncJointFormTooManyArgs(t *testing.T) {
-	tf, cmd := prepareCompletionTest()
-	pods, _, _ := cmdtesting.TestData()
-	addResourceToFactory(tf, pods)
+	testCases := []struct {
+		name              string
+		obj               runtime.Object
```

Comment:
It seems you only using `barPod` as obj, is there any reason to add obj in test table?

Reply:
Updated thanks!